app/models/miq_vim_broker_worker.rb

```diff
@@ -1,13 +1,17 @@
 class MiqVimBrokerWorker < MiqWorker
   require_nested :Runner
 
-  self.required_roles         = %w(
-    ems_inventory
-    ems_metrics_collector
-    ems_operations
-    smartproxy
-    smartstate
-  )
+  self.required_roles         = lambda {
+    %w(
+      ems_metrics_collector
+      ems_operations
+      smartproxy
+      smartstate
+    ).tap do |roles|
+      roles << 'ems_inventory' unless Settings.prototype.try(:ems_vmware).try(:update_driven_refresh)
```

Comment:
I'm confused why you have to do this as well as check the setting and role in the `has_ems_inventory_role?` method in the runner.